front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -663,10 +663,21 @@ export function DataSourceViewSelector({
   const isExpanded = searchResult
     ? searchResult.dataSourceView.sId === dataSourceView.sId
     : false;
-  const defaultExpandedIds =
-    isExpanded && searchResult
-      ? removeNulls([...new Set(searchResult.parentInternalIds)])
-      : undefined;
+
+  const defaultExpandedIds = useMemo(
+    () =>
+      searchResult && isExpanded
+        ? removeNulls([
+            ...new Set(
+              searchResult.parentInternalIds?.filter(
+                (id) =>
+                  searchResult.expandable || id !== searchResult.internalId
```

Comment:
do we really need the `searchResult.expandable` ? do we want to expand the node itself , even if expandable ?

Reply:
I guess we wanted to do it looking at the original code? I can ask Ed what he thinks

Reply:
Yes he said keep as is!